Goal: Do not use PAM for chage, chpasswd, groupadd, groupdel, groupmod
      newusers, useradd, userdel, usermod, chgpasswd (keep them low-level)
Fixes: #283961, #162181, #162199, #162228, #369806.

Status wrt upstream: A configure option was added. To be tested with 4.1.3.

Index: shadow-4.1.2/src/chage.c
===================================================================
--- shadow-4.1.2.orig/src/chage.c	2008-06-13 23:23:11.000000000 +0200
+++ shadow-4.1.2/src/chage.c	2008-06-16 02:29:28.463262026 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: chage.c 2162 2008-06-13 21:23:09Z nekral-guest $"
 
@@ -53,6 +54,7 @@
 #include "exitcodes.h"
 #include "prototypes.h"
 #include "defines.h"
+#undef USE_PAM
 #include "pwio.h"
 #include "shadowio.h"
 /*
Index: shadow-4.1.2/src/chpasswd.c
===================================================================
--- shadow-4.1.2.orig/src/chpasswd.c	2008-06-13 20:11:10.000000000 +0200
+++ shadow-4.1.2/src/chpasswd.c	2008-06-16 02:30:16.079421379 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: chpasswd.c 2130 2008-06-13 18:11:09Z nekral-guest $"
 
@@ -43,6 +44,7 @@
 #include "pam_defs.h"
 #endif				/* USE_PAM */
 #include "defines.h"
+#undef USE_PAM
 #include "exitcodes.h"
 #include "nscd.h"
 #include "prototypes.h"
Index: shadow-4.1.2/src/groupadd.c
===================================================================
--- shadow-4.1.2.orig/src/groupadd.c	2008-06-13 23:19:12.000000000 +0200
+++ shadow-4.1.2/src/groupadd.c	2008-06-16 02:30:41.299262034 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: groupadd.c 2161 2008-06-13 21:19:07Z nekral-guest $"
 
@@ -47,6 +48,7 @@
 #endif				/* USE_PAM */
 #include "chkname.h"
 #include "defines.h"
+#undef USE_PAM
 #include "getdef.h"
 #include "groupio.h"
 #include "nscd.h"
Index: shadow-4.1.2/src/groupdel.c
===================================================================
--- shadow-4.1.2.orig/src/groupdel.c	2008-06-13 22:59:07.000000000 +0200
+++ shadow-4.1.2/src/groupdel.c	2008-06-16 02:30:49.651262093 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: groupdel.c 2154 2008-06-13 20:59:04Z nekral-guest $"
 
@@ -44,6 +45,7 @@
 #include <stdio.h>
 #include <sys/types.h>
 #include "defines.h"
+#undef USE_PAM
 #include "groupio.h"
 #include "nscd.h"
 #include "prototypes.h"
Index: shadow-4.1.2/src/groupmod.c
===================================================================
--- shadow-4.1.2.orig/src/groupmod.c	2008-06-13 23:08:07.000000000 +0200
+++ shadow-4.1.2/src/groupmod.c	2008-06-16 02:30:58.227367347 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: groupmod.c 2158 2008-06-13 21:08:06Z nekral-guest $"
 
@@ -46,6 +47,7 @@
 #endif				/* USE_PAM */
 #include "chkname.h"
 #include "defines.h"
+#undef USE_PAM
 #include "groupio.h"
 #include "pwio.h"
 #include "nscd.h"
Index: shadow-4.1.2/src/newusers.c
===================================================================
--- shadow-4.1.2.orig/src/newusers.c	2008-06-13 22:56:45.000000000 +0200
+++ shadow-4.1.2/src/newusers.c	2008-06-16 02:31:41.669480684 +0200
@@ -39,6 +39,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: newusers.c 2152 2008-06-13 20:56:44Z nekral-guest $"
 
@@ -55,6 +56,7 @@
 #endif				/* USE_PAM */
 #include "prototypes.h"
 #include "defines.h"
+#undef USE_PAM
 #include "getdef.h"
 #include "groupio.h"
 #include "nscd.h"
Index: shadow-4.1.2/src/useradd.c
===================================================================
--- shadow-4.1.2.orig/src/useradd.c	2008-06-16 01:33:59.000000000 +0200
+++ shadow-4.1.2/src/useradd.c	2008-06-16 02:31:09.347908114 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: useradd.c 2189 2008-06-15 22:25:51Z nekral-guest $"
 
@@ -51,6 +52,7 @@
 #include <time.h>
 #include "chkname.h"
 #include "defines.h"
+#undef USE_PAM
 #include "faillog.h"
 #include "getdef.h"
 #include "groupio.h"
Index: shadow-4.1.2/src/userdel.c
===================================================================
--- shadow-4.1.2.orig/src/userdel.c	2008-06-13 22:23:28.000000000 +0200
+++ shadow-4.1.2/src/userdel.c	2008-06-16 02:31:15.268028621 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: userdel.c 2147 2008-06-13 20:23:26Z nekral-guest $"
 
@@ -47,6 +48,7 @@
 #include "pam_defs.h"
 #endif				/* USE_PAM */
 #include "defines.h"
+#undef USE_PAM
 #include "getdef.h"
 #include "groupio.h"
 #include "nscd.h"
Index: shadow-4.1.2/src/usermod.c
===================================================================
--- shadow-4.1.2.orig/src/usermod.c	2008-06-16 01:52:28.000000000 +0200
+++ shadow-4.1.2/src/usermod.c	2008-06-16 02:31:21.364381584 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: usermod.c 2151 2008-06-13 20:40:04Z nekral-guest $"
 
@@ -50,6 +51,7 @@
 #include <time.h>
 #include "chkname.h"
 #include "defines.h"
+#undef USE_PAM
 #include "faillog.h"
 #include "getdef.h"
 #include "groupio.h"
Index: shadow-4.1.2/src/chgpasswd.c
===================================================================
--- shadow-4.1.2.orig/src/chgpasswd.c	2008-06-13 20:11:10.000000000 +0200
+++ shadow-4.1.2/src/chgpasswd.c	2008-06-16 02:31:48.547282333 +0200
@@ -31,6 +31,7 @@
  */
 
 #include <config.h>
+#undef USE_PAM
 
 #ident "$Id: chgpasswd.c 2130 2008-06-13 18:11:09Z nekral-guest $"
 
@@ -43,6 +44,7 @@
 #include "pam_defs.h"
 #endif				/* USE_PAM */
 #include "defines.h"
+#undef USE_PAM
 #include "nscd.h"
 #include "prototypes.h"
 #include "groupio.h"
